Source/WebKit/Platform/IPC/Connection.h

```diff
@@ -1106,5 +1127,34 @@ inline void markCurrentlyDispatchedMessageAsInvalid(const RefPtr<Connection>& co
         connection->markCurrentlyDispatchedMessageAsInvalid();
 }
 
+template<typename T>
+inline std::enable_if_t<std::is_same_v<T, Connection>> markCurrentlyDispatchedMessageAsInvalid(const Ref<T>& connection)
+{
+    connection->markCurrentlyDispatchedMessageAsInvalid();
+}
+
+#if ENABLE(IPC_TESTING_API)
```

Comment:
To prevent proliferation of these functions, could you perhaps just add the string as a parameter to markCurrentlyDispatchedMessageAsInvalid?